File: src/test/java/org/dedda/bratwurst/lang/ExitTest.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.ScopedTestCase;
import org.dedda.bratwurst.lang.scope.Scope;
import org.testng.annotations.Test;

import static org.testng.Assert.assertFalse;
import static org.testng.Assert.assertTrue;

/**
 * Created by dedda on 12/17/15.
 *
 * @author dedda
 */
public class ExitTest extends ScopedTestCase {

    private Scope scope;

    public void setUp() throws Exception {
        scope = createEmptyScope();
        scope.getProgram().run();
    }

    @Test
    public void testRun() throws Exception {
        Exit exit = new Exit(0);
        assertFalse(scope.getProgram().isStopped());
        exit.run(scope);
        assertTrue(scope.getProgram().isStopped());
    }
}

File: src/test/java/org/dedda/bratwurst/lang/ExitTest.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.ScopedTestCase
import org.dedda.bratwurst.lang.scope.Scope
import org.testng.Assert
import org.testng.annotations.Test

/**
 * Created by dedda on 12/17/15.
 *
 * @author dedda
 */
class ExitTest : ScopedTestCase() {
    private lateinit var scope: Scope

    @Throws(Exception::class)
    public override fun setUp() {
        scope = createEmptyScope()
        scope.program?.run()
    }

    @Test
    @Throws(Exception::class)
    fun testRun() {
        val exit = Exit(0)
        Assert.assertFalse(scope.program.isStopped)
        exit.run(scope)
        Assert.assertTrue(scope.program.isStopped)
    }
}